src/vivarium/framework/resource.py

```diff
@@ -335,7 +340,7 @@ def add_resources(
         """
         self._manager.add_resources(resource_type, resource_names, producer, dependencies)
 
-    def __iter__(self):
+    def __iter__(self) -> Iterator[Callable[..., Any]]:
```

Comment:
I think this is correct but I'd like to talk about this w/ someone familiar with python's implementation of iterables